mobile/lib/shared/services/api.service.dart

```diff
@@ -95,14 +97,19 @@ class ApiService {
       serverUrl += '/api';
     }
 
-    // Throw Socket or Timeout exceptions,
-    // we do not care if the endpoints hits an HTTP error
     try {
-      await client
+      final response = await client
           .get(
             Uri.parse(serverUrl),
           )
           .timeout(const Duration(seconds: 5));
+
+      if (response.statusCode == 502) {
```

Comment:
You should probably do the opposite actually. If it isn't a 200, log the error and return false.

Reply:
Done; it never hit a valid GET endpoint, so it returned 404 or 512. I modify to use GET /server-info/ping to get a valid HTTP 200 when possible

Reply:
Will $serverUrl contain the /api suffix even if not provided on the login screen? I know it works without it currently but im not sure where that gets added/checked.

Reply:
@mmomjian it gets added if it is not included